orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/QueueShovel.kt

```diff
@@ -81,7 +90,33 @@ class QueueShovel(
     }
   }
 
+  private fun transferOwnership(message: Message) {
+    if (executionRepository == null) {
+      return
+    }
+
+    if (message !is ExecutionLevel) {
+      log.warn("Message $message does not implement ExecutionLevel, can not inspect partition")
+      return
+    }
+
+    // don't catch exceptions on retrieve/store (e.g. ExecutionNotFoundException), so that we can short-circuit shoveling
+    // of this message
```

Comment:
If the execution isn't found, we'll log an error and then do nothing with the message: Won't we just enter into a loop of logging out an error for N messages, but never actually doing anything with them?

Reply:
Yes, another way to think about it is that we'll keep retrying shoveling that message. It may eventually succeed, for instance if the execution was missing because of peering lag

Reply:
but won't the message have been taken and ack'd already so, in essence, we would just dump it on the floor in the peering lag case?

Reply:
@marchello2000 not sure what you mean, I don't see how we dump anything on the floor? There is a test that makes sure we don't ack messages if there is a problem reading/updating their execution